src/frontend/Ast_to_Mir.ml

```diff
@@ -679,6 +679,22 @@ let migrate_checks_to_end_of_block stmts =
   let checks, not_checks = List.partition_tf ~f:stmt_contains_check stmts in
   not_checks @ checks
 
+let trans_data (p : Ast.typed_program) =
```

Comment:
To me the name `trans_data` implies this will be part of `trans_prog` which I think is a little confusing. Maybe `gather_data`?